src/components/layout/ChatArea.tsx

```diff
@@ -407,11 +429,59 @@ export const ChatArea: React.FC<{
     toggleMemberList,
     openPrivateChat,
     messages,
+    connect,
+    joinChannel,
+    toggleAddServerModal,
   } = useStore();
 
   // Tab completion hook
   const tabCompletion = useTabCompletion();
 
+  const handleIrcLinkClick = (rawUrl: string) => {
+    // Tolerate trailing punctuation in chat text
+    const sanitized = rawUrl.trim().replace(/[),.;:]+$/, "");
+    const urlObj = new URL(sanitized);
+    const host = urlObj.hostname;
+    const scheme = urlObj.protocol.replace(":", "");
+    const port = urlObj.port
```

Comment:
How do we distinct TCP from websocket... when it is stupported if it is ever supported. Technically we are connecting to ws://irc.something.com or wss://irc.something.com, because the default irc ports would be 6667 and 6697. If we say the native clients should support these urls they probably should use the TCP ones when that is supported, as most urls like this would have a TCP one but not necessarily websocket. Uff not sure what to do here.

Reply:
But the web version will never support TCP, is only the native, and if native URL handling is different from web is weird. We could have a bouncer but still the web version should run without a bouncer...